src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -83,12 +83,20 @@ public function getLogDir(): string
         return $_SERVER['APP_LOG_DIR'] ?? parent::getLogDir();
     }
 
+    /**
+     * Gets the path to the bundles configuration file.
+     */
+    public function getBundlesPath(): string
```

Comment:
the method should be private I suppose (and listed after public ones then)

Reply:
May be protected? It should be overridable.

Reply:
a private method is a trait can be overriden when importing the trait, isn't that enough?

Reply:
I think this approach is inconvenient.
Yes, private method can be overridden, but if you use some smart IDE (e.g. PhpStorm) this will lead at least 2 disadvantages:
1. "Override Methods" action will not suggest you this method.
2. Overridden private method will be highlighted as unused.

Using of protected method in such case is more clear for me. Considering that it will need to be described in the docs, I think this is important.

Reply:
if PhpStorm reports that private method as unused, then it is a bug in PhpStorm not being smart enough. That's where it should be reported

Reply:
Maybe you are right about a bug in PhpStorm, but your suggestion is making the DX worse. I see the following traits that using the protected methods:
`Symfony\Component\HttpClient\Response\TransportResponseTrait`
`Symfony\Component\Routing\Loader\Configurator\Traits\PrefixTrait`
`Symfony\Component\Routing\Loader\Configurator\Traits\HostTrait`
`Symfony\Component\Routing\Loader\Configurator\Traits\LocalizedRouteTrait`
`Symfony\Component\VarDumper\Test\VarDumperTestTrait`
`Symfony\Component\Cache\Tests\Adapter\PdoPruneableTrait`
`Symfony\Component\Cache\Traits\FilesystemTrait`
`Symfony\Component\Cache\Traits\FilesystemCommonTrait`
`Symfony\Component\Cache\Traits\RedisTrait`
`Symfony\Component\Cache\Traits\MemcachedTrait`
`Symfony\Component\Form\Test\Traits\ValidatorExtensionTrait`
`Symfony\Component\Serializer\Normalizer\ObjectToPopulateTrait`
`Symfony\Bundle\FrameworkBundle\Command\BuildDebugContainerTrait`
and some of them has the both protected and private methods (e.g. TransportResponseTrait).

If there is some logic when we should use a protected method in a trait and when a private ones, I would be thankful if you could explain it.

Reply:
I think I understood difference between protected and private methods in traits and the previous question can be considered resolved.